crates/electrum/src/bdk_electrum_client.rs

```diff
@@ -371,120 +287,90 @@ impl<E: ElectrumApi> BdkElectrumClient<E> {
-                None => continue,
-            };
+                self.validate_merkle_for_anchor(graph_update, txid, r.height)?;
+            }
 
             let _ = graph_update.insert_tx(tx);
-            if let Some(anchor) = anchor {
-                let _ = graph_update.insert_anchor(txid, anchor);
-            }
         }
         Ok(())
     }
-}
-
-/// The result of [`BdkElectrumClient::full_scan`].
-///
-/// This can be transformed into a [`FullScanResult`] with either [`ConfirmationHeightAnchor`] or
-/// [`ConfirmationTimeHeightAnchor`] anchor types.
-pub struct ElectrumFullScanResult<K>(FullScanResult<K, ConfirmationHeightAnchor>);
-
-impl<K> ElectrumFullScanResult<K> {
-    /// Return [`FullScanResult`] with [`ConfirmationHeightAnchor`].
-    pub fn with_confirmation_height_anchor(self) -> FullScanResult<K, ConfirmationHeightAnchor> {
-        self.0
-    }
-
-    /// Return [`FullScanResult`] with [`ConfirmationTimeHeightAnchor`].
-    ///
-    /// This requires additional calls to the Electrum server.
-    pub fn with_confirmation_time_height_anchor(
-        self,
-        client: &BdkElectrumClient<impl ElectrumApi>,
-    ) -> Result<FullScanResult<K, ConfirmationTimeHeightAnchor>, Error> {
-        let res = self.0;
-        Ok(FullScanResult {
-            graph_update: try_into_confirmation_time_result(res.graph_update, &client.inner)?,
-            chain_update: res.chain_update,
-            last_active_indices: res.last_active_indices,
-        })
-    }
-}
-
-/// The result of [`BdkElectrumClient::sync`].
-///
-/// This can be transformed into a [`SyncResult`] with either [`ConfirmationHeightAnchor`] or
-/// [`ConfirmationTimeHeightAnchor`] anchor types.
-pub struct ElectrumSyncResult(SyncResult<ConfirmationHeightAnchor>);
 
-impl ElectrumSyncResult {
-    /// Return [`SyncResult`] with [`ConfirmationHeightAnchor`].
-    pub fn with_confirmation_height_anchor(self) -> SyncResult<ConfirmationHeightAnchor> {
-        self.0
+    // Helper function which checks if a transaction is confirmed by validating the merkle proof.
+    // An anchor is inserted if the transaction is validated to be in a confirmed block.
+    fn validate_merkle_for_anchor(
+        &self,
+        graph_update: &mut TxGraph<ConfirmationTimeHeightAnchor>,
+        txid: Txid,
+        confirmation_height: i32,
+    ) -> Result<(), Error> {
+        if let Ok(merkle_res) = self
+            .inner
+            .transaction_get_merkle(&txid, confirmation_height as usize)
+        {
+            let header = self.inner.block_header(merkle_res.block_height)?;
+            let is_confirmed_tx = electrum_client::utils::validate_merkle_proof(
+                &txid,
+                &header.merkle_root,
+                &merkle_res,
+            );
+
+            if is_confirmed_tx {
+                let _ = graph_update.insert_anchor(
+                    txid,
+                    ConfirmationTimeHeightAnchor {
```

Comment:
This seems like this is a suboptimal anchor to use since you're entering the same `block_height` twice. It would actually be good if you could delete `ConfirmationTimeHeightAnchor` and `ConfirmationHeightAnchor` from the codebase now since both electrum and esplora will report the exact block that the tx is in with this PR. It only needs to be recorded once.

For now this would be good to use for both:

```rust
struct ConfirmationBlockTime {
    block_id: BlockId,
    confirmation_time: u64
}
```

Later we can add the block position to this and maybe even the proof. 

Deleting `ConfirmationTimeHeight` anchor would need ACK from @evanlinjin  and @notmandatory since it's used by `Wallet`. Just for background the only reason `ConfirmationTimeHeightAnchor` exists is because of how wacky our electrum client was before this PR. I think simplifying the anchors to make use of this should be done ASAP.

Reply:
If we don't want to delete the old anchors we should still return the correct thing from here. It can be `Into` converted into the old one later.

Reply:
ACK

Reply:
I support simplifying the anchors now. I'm happy to do whatever changes are required for to the sqlite storage crate. 

cc: @matthiasdebernardini this anchor change will also impact the postgres store you're working on.